### DISCUSSION TITLE
Switch from personal (education) Github Copilot to organization Copilot

### DISCUSSION THREAD

**Discussion by KhanhhNe** [community/community#103205] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hi everyone,

I am currently enrolling in Github Student Developer Program, which includes a Github Copilot subscription. I joined an organization which requires any Github Copilot usage must be using their own subscription (with their policies etc), and personal subscription is not allowed.

I cannot see anywhere in both my Billing and Github Copilot page specifying an option to switch to the organization's subscription. How can I verify which one am I using, and how to switch to the organization Copilot if I'm using the one provided by Github Student Pack?

**Comment by davecheney** [comment, marked as answer]
Once your employer sends you an invitation to join their Copilot enabled organisation, your personal subscription will be cancelled. You will receive an email when that occurs.

**Reply by KhanhhNe** [reply]
thank you so much!

**Reply by Vince-Smith** [reply]
Wait, so there's no way to switch between my personal co-pilot subscription and my organizations?

What if my organization has hard privacy/model limitations that I might not want in place when performing my personal work?

**Reply by paladini** [reply]
It's insane that I cannot use my own student Github Copilot license to implement exercises and open-source projects, and when at work, switch to use my company's Github Copilot license. 

So If I need to code something for the university, I need to use my Premium Requests from my company. But I do have Github Student Pack for that purpose - for studying! The company should not be affected by my personal life, as well I would like to train Copilot with my student code usage.

**Reply by marcovoliveira** [reply]
Fix this! I want to switch between student and organization

**Reply by K-Schaeffer** [reply]
Ok I'm having a lot of issues with this, specially because I ran out of premium requests while working on study projects and that ended up using my org requests instead of my student ones.

It is really important to be able to switch this.

**Reply by Haprog** [reply]
As an employee of a consultancy working for multiple large customer organizations, I have access to multiple GitHub enterprise organizations which provide Copilot licenses and each organization might have different settings and policies for what is enabled etc.

It would also be very beneficial to be able to switch between which organizations license I'm using depending on what project I'm working on. Currently it seems like the only workaround for this would be to use multiple GitHub accounts (e.g. one for personal projects and one for each employer/customer organization).

**Reply by shnigi** [reply]
> As an employee of a consultancy working for multiple large customer organizations, I have access to multiple GitHub enterprise organizations which provide Copilot licenses and each organization might have different settings and policies for what is enabled etc.
> 
> It would also be very beneficial to be able to switch between which organizations license I'm using depending on what project I'm working on. Currently it seems like the only workaround for this would be to use multiple GitHub accounts (e.g. one for personal projects and one for each employer/customer organization).

And I believe creating multiple personal github accounts is against the github policy so it can't be done.  At least as of february 2026 it is now possible to change the billing organization from github settings which is a step forward. However it can't be selected from VScode directly and the organization settings can't be selected either.

**Comment by ceejboy** [comment]
![image](https://github.com/user-attachments/assets/ebfffbd6-287e-4de4-965f-bad028624938)

I receive this, but can't see the copilot subscription on my end

**Comment by airtonix** [comment]
Fail.

Epic Fail.


While working on stuff at work, I just realised all these premium requests have been blowing away my personal pro account quota.

We need a way to choose.

**Comment by MattBonacini** [comment]
So 2 years later this isn't yet solved. I don't have an education plan, but need separate plans for my side projects and work. Now I'm forced to create 2 Github accounts and manage both, which means I need to manage also 2 different versions of vs code, and for sure more work to auth via terminal commands too

**Comment by redsquare** [comment]
This is impacting me badly too, why cannot I simply sign into the correct context in vscode?

**Comment by SergiusTheBest** [comment]
I had a personal subscription with a lot of models enabled and other experimental stuff and today one of my organizations granted me their business copilot subscription (lucky me!) and all models are gone now and most of the features are turned off with no way to enable them :( We definitely need a way to switch between organizations without creating a separate account for each of them.

**Comment by Cypherball** [comment]
If you guys at GitHub aren't going to separate out subscriptions for Org and Personal, at least have distinct permissions for models at the org level and user level. I don't want the same restrictions to apply while working on my personal projects :/

**Comment by danabadarneh** [comment]
Thanks for the clear explanation and the helpful answer.
This is a very common situation for students joining organizations, and your response makes it much easier to understand how Copilot subscriptions work between personal and organization accounts. Really appreciate the clarification!

**Comment by Evander-Greer** [comment]
I saw your post and you don't have to worry about that.
Here are the most exact solution:
The user is enrolled in the GitHub Student Developer Program, which includes a GitHub Copilot subscription. However, they’ve joined an organization that requires members to use the organization's GitHub Copilot subscription (with specific policies) and does not allow personal subscriptions.

They are asking how to verify which subscription they are currently using (the personal GitHub Copilot subscription from the GitHub Student Pack or the organization’s subscription) and how to switch to the organization’s Copilot subscription.

Answer and Steps to Resolve the Issue:
1. Verify the Current GitHub Copilot Subscription:

You can verify which GitHub Copilot subscription you're using by checking the Billing & Subscription page in your GitHub account settings.

Go to your GitHub account and navigate to Settings.

Click on Billing & Plans (under Billing).

Here, you will be able to see whether your GitHub Copilot subscription is tied to your personal account (via the GitHub Student Pack) or the organization’s subscription.

You should see the following:

If you're using GitHub Copilot from the Student Developer Program, it will likely show that you have GitHub Copilot for Students activated.

If you're using the organization’s subscription, it will show the subscription tied to the organization.

2. Switching to the Organization’s Copilot Subscription:

You cannot simply "switch" between subscriptions in the GitHub Copilot settings. If the organization requires you to use their GitHub Copilot subscription, there are a few steps to take:

Leave your personal GitHub Copilot subscription:

To ensure you’re using the organization’s Copilot subscription, you may need to cancel or disable your personal GitHub Copilot subscription.

In the Billing & Plans section, you should find an option to manage your Copilot subscription, which may allow you to cancel your personal Copilot (if it's linked to your GitHub Student Pack).

Join the organization’s Copilot plan:

Your organization likely has a specific GitHub Enterprise plan or another Copilot plan that you need to be added to. You may need to contact your organization's GitHub administrator or owner to confirm that they have assigned you the correct Copilot subscription for organization use.

The organization administrator can assign you a seat under their plan, and this will automatically replace your personal subscription.

Verify with your organization:

Check with your GitHub organization administrator to ensure they’ve correctly set up GitHub Copilot access under their subscription.

You might need to log out and log back in to GitHub for the change to take effect after they assign the subscription.

3. Confirming Copilot Subscription Usage:

Once the subscription has been updated or switched, you can confirm which subscription is active by:

Checking GitHub’s Copilot page: Visit [GitHub Copilot’s settings page](https://github.com/settings/copilot)
 to see what plan is currently active for your account.

Confirm via the organization’s settings: Your organization’s admin should have visibility into your Copilot subscription status, and they can verify that you’re using the correct subscription.
Best regards.

**Comment by KennethhSWE** [comment]
Tried to do the same as a Student at ASU and I just gave up after awhile.

**Comment by Evander-Greer** [comment]
I’m looking to connect with a partner who brings strong business and
technology consulting expertise.
I have over 7 years of experience working as a web developer on major
freelancing platforms such as Upwork and Toptal, with a strong track record
and extensive GitHub contributions.
Due to account issues related to platform policies, I am now looking to
build a stable long-term partnership.
The goal is to operate as a private development team, where you act as the
public-facing profile in the U.S., and I handle the technical development
work in the background.
This setup is important because the U.S. market is the largest freelancing
market, and having a U.S.-based account and profile significantly increases
trust, client access, and earning potential.
I will be responsible for development and technical delivery, while you
support with client communication, interviews, and basic project
coordination when needed.

Ownership & Payment Revenue split: 30% (you)
Monthly base: $300, increasing by $100 every 3 months, up to $1,000
Interview & business support:$80/hr$150/hr
if an interview leads to a successful hire
Requirements U.S.-based profile and account
setup Stable laptop, internet, and power
Availability for occasional client interviews or meetings
Payment setup via PayPal (linked to your bank or card)

This partnership is designed to be long-term, reliable, and mutually
beneficial, focused on growing consistent freelance and agency-style income
in the U.S. market.

**Comment by khaledosman** [comment]
Fun fact, if you're in an org copilot plan and you reach your limit, you cant switch or use a personal plan or pay extra credits yourself, you need an org admin to remove your seat from the org.. which only takes effect after the end of the month, so you pretty much can't use copilot until the end of the month.. Only after that period can you switch back to an individual plan to be able to use copilot again or upgrade, epic fail indeed.

**Reply by Vince-Smith** [reply]
That IS fun!

**Comment by PerfectlyFineCode** [comment]
Fix this!!

**Comment by Alphayeeeet** [comment]
Fix this!
